src/kvstore/kvstore_local.h

```diff
@@ -174,13 +175,75 @@ class KVStoreLocal : public KVStore {
 
  protected:
   /**
-   * \brief group values on keys
+   * \brief group values on keys for push
    */
-  template <typename V>
+  void GroupKVPairsPush(const std::vector<int>& keys,
```

Comment:
A second thought is defining validator lambda function in the caller functions or a common place accessible to various callers and pass it to `GroupKVPairs`. In this way, there is no need to define extra interfaces of GroupKVPairsXXXX where XXXX stands for Push and Pull, respectively. Is this feasible?